libcalico-go/lib/backend/k8s/resources/workloadendpoint.go

```diff
@@ -53,6 +53,27 @@ type WorkloadEndpointClient struct {
 	converter conversion.Converter
 }
 
+type patchModeKey struct{}
```

Comment:
I might suggest putting this into a new file rather than workloadendpoint.go, to help reduce the risk of merge conflits.

Reply:
Are you thinking just the `patchModeKey` or that more of the changes in this PR should move to a new file?

Reply:
I was thinking this whole section - lines 56 to 75.

Mostly because I know there are differences in this file and smaller more focused files are easier to manage conflicts in.

Reply:
Sounds good, I separated it out.